docs/developers/meta/sc-meta-cli.md

```diff
@@ -0,0 +1,373 @@
+- `sc-meta all build --locked` will build all contracts, with the crate versions given by Cargo.lock;
+- `sc-meta all clean` will clean all projects;
+- etc.
+
+You can call `sc-meta all help` and see that the CLI docs are almost the same as those of the individual contract tool.
+
+A related command is the `info` command, which just prints a tree with all the contract and contract libraries in a folder, without doing anything to them.
+
+:::info Note
+Since the CLI of the individual contract tool is available in the standalone tool, the following two commands are equivalent:
+
+```
+cd my-contract
+sc-meta all build
+```
+
+```
+cd my-contract/meta
+cargo run build
+```
+:::
+
+Paramameters:
+- `--path`
+    - Target directory where to call all contract meta crates.
+    - _default_: current directory.
+- `--ignore`
+    - Ignore all directories with these names.
+    - _default_: `target`.
+- `--no-abi-git-version
+    - Skips loading the Git version into the ABI
+- `--target-dir-meta`
+    - For the meta crates, allows specifying the target directory where the Rust compiler will build the intermediary files. Sharing the same target directory can speed up building multiple contract crates at once.
+    - _default_: uses the workspace, or builds in `meta/target`.
+- `--target-dir-all`
+    - Overrides both the `--target-dir-meta` and the `--target-dir-wasm` args.
+
+
+[comment]: # (mx-context-auto)
+
+### Calling `info`
+
+The info command prints an overview of the contracts and libraries and residing under a folder. It also prints their framework versions.
+
+As an example, below is the output of calling it in the example contract folder in the framework:
+
+![sc-meta info screenshot](/developers/sc-meta/sc-meta-info.png "Result of calling sc-meta info in the example contract folder in the framework")
+
+Paramameters:
+- `--path`
+    - Target directory where to call all contract meta crates.
+    - _default_: current directory.
+- `--ignore`
+    - Ignore all directories with these names.
+    - _default_: `target`.
+
+
+[comment]: # (mx-context-auto)
+
+### Calling `upgrade`
+
+Calling `sc-meta upgrade` will try to automatically upgrade a contract or group of contracts to the latest version.
+
+The oldest version currently supported is `0.28.0`. Any older than that, and the developer will need to upgrade it by hand to `0.28.0`.
+
+It is especially important when upgrading from `0.38` to `0.39.0`, since a lot of changes happened at that point.
+
+:::tip
+For projects with multiple contract crates, we recommend upgrading all of them at once. The upgrade algorithm goes step by step, version after version. For some of the major versions, it also checks that the project compiles before moving on. This is to give developers the chance to fix issues manually, if necessary, and not have those issues pile up. If there are local depdencies between contracts, the upgrader will not be able to do the check unless all of them are upgraded together.
+:::
+
```

Comment:
```suggestion

:::caution
Generally, we strongly recommend to ensure code versioning or at least a backup of the contract code to avoid the impossibility of reverting permanent changes. This automatic code altering process involved in using `sc-meta upgrade` highly raises this recommendation.
:::

```